pkg/apis/wsecurity/v1/count.go

```diff
@@ -0,0 +1,146 @@
+package v1
+
+import (
+	"bytes"
+	"fmt"
+)
+
+//////////////////// CountProfile ////////////////
+
+// Exposes ValueProfile interface
+type CountProfile uint8
+
+func (profile *CountProfile) Profile(args ...interface{}) {
+	*profile = CountProfile(args[0].(uint8))
+}
+
+func (profile *CountProfile) String(depth int) string {
+	return fmt.Sprintf("%d", uint8(*profile))
+}
+
+//////////////////// CountPile ////////////////
+
+// Exposes ValuePile interface
+type CountPile []uint8
+
+func (pile *CountPile) Add(valProfile ValueProfile) {
+	profile := *valProfile.(*CountProfile)
+	*pile = append(*pile, uint8(profile))
+}
+
+func (pile *CountPile) Clear() {
+	*pile = nil
+}
+
+func (pile *CountPile) Merge(otherValPile ValuePile) {
+	otherPile := otherValPile.(*CountPile)
+	*pile = append(*pile, *otherPile...)
+}
+
+func (pile *CountPile) String(depth int) string {
+	return fmt.Sprintf("%v", *pile)
+}
+
+//////////////////// CountConfig ////////////////
+type countRange struct {
+	Min uint8 `json:"min"`
+	Max uint8 `json:"max"`
+}
+
+func (cRange *countRange) fuseTwoRanges(otherRange *countRange) bool {
+	if cRange.Max < otherRange.Min || cRange.Min > otherRange.Max {
+		// no overlap - nothing to do!
+		return false
+	}
+
+	// There is overlap of some sort
+	if cRange.Min > otherRange.Min {
+		cRange.Min = otherRange.Min
+	}
+	if cRange.Max < otherRange.Max {
+		cRange.Max = otherRange.Max
+	}
+	return true
+}
+
+// Exposes ValueConfig interface
+type CountConfig []countRange
+
+func (config *CountConfig) Decide(valProfile ValueProfile) string {
```

Comment:
That will indeed be more aligned with go standard  
However, it will make it harder to accumulate multiple issues found in a tree of Data Types. 

I think we need to consider this in the next PR when we see a Data Type that relies on count (depth of 2), 
Ideally, we will define a mechanism that will allow us to accumulate issues (not great with errors, very poor but can be used on day 1 with strings)

Reply:
Take a look at the pattern here for errors: https://github.com/knative/pkg/blob/main/apis/field_error.go#L63

In particular, you can define an accumulator for errors like so: https://github.com/knative/pkg/blob/main/apis/field_error.go#L196

```golang
func (fe *FieldError) Also(errs ...*FieldError) *FieldError {
  // Note that `fe` may be `nil` here
```

Reply:
So, error accumulation may be another option. 

Let's delay this discussion to the next PR though where we get to see 2 levels of data types.

We need to discuss if to accumulate and what is the best way.

Reply:
My only worry in waiting is if we change the output of an exported v1 API, does that require a new v2 version?  I know it'll likely have little impact right now (so I may be over thinking this), but technically v1 does require backwards compatibility so would this be considered a "breaking" change...

Reply:
We are long before our Guard v0.1
As for API version, I did not answer the above since I am not sure how to relate the API version to the Guard version. 
I think we will have many Guard versions and potentially more than one API version - and I am not sure what the correct practice is as for API version - will try to read about that and check if we can move to a v0.1 API version

Reply:
in that case, should we change the path so that it's not `.../v1/...`? I think that was the main thing that was hanging me up

Reply:
I'd remove the version number for the v0.x version. You can add the version numbering and enjoy `go mod` versioning once you have a breaking change _after_ v1.

Reply:
We moved to v1aplha1 as customary in Kube.